### PR TITLE
React: Remove deprecated material-ui code

### PR DESCRIPTION
**Pull request by hannahle**
Resolved Material-UI related deprecation warnings

### CODE REVIEW

File: react/src/components/ListItemRouterLink.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default function ListItemRouterLink(props: ListItemRouterLinkProps) {
                     {...linkProps}
                     onMouseEnter={() => setOpenTooltip(true)}
                     onMouseLeave={() => setOpenTooltip(false)}
-                />
+                ></Link>
```
Comment (cklamann) [inline react/src/components/ListItemRouterLink.tsx:26]: Is this change necessary?

Comment (hannahle) [reply react/src/components/ListItemRouterLink.tsx:26]: I think there was a deprecation warning that requires Link to have children nodes. It's totally fine if we remove it though since we don't have any children to begin with!

Comment (cklamann) [reply react/src/components/ListItemRouterLink.tsx:26]: OK sounds good -- the children should be passed through the props, so hopefully it doesn't complain if you revert!

**Review by frewmack — COMMENTED** [review]
Is this PR for issue #752? The migration guide seems to provide some [codemod scripts](https://next.material-ui.com/guides/migration-v4/#run-codemods) that we can run on the `/react` folder to automatically make changes:

```
npx @material-ui/codemod@next v5.0.0/preset-safe <folder>
npx @material-ui/codemod@next v5.0.0/variant-prop <folder>
npx @material-ui/codemod@next v5.0.0/link-underline-hover <folder>
```

**Review by frewmack — APPROVED** [review]
Looks good to me! No warnings on my end
